types/src/transaction/transaction.rs

```diff
@@ -51,10 +51,52 @@ pub struct AssetOutPoint {
 #[serde(rename_all = "camelCase")]
 pub struct AssetTransferInput {
     pub prev_out: AssetOutPoint,
+    pub timelock: Option<Timelock>,
     pub lock_script: Bytes,
     pub unlock_script: Bytes,
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase", tag = "type", content = "value")]
+pub enum Timelock {
+    Block(u64),
+    BlockAge(u64),
+    Time(u64),
+    TimeAge(u64),
+}
+
+type TimelockType = u8;
+const BLOCK: TimelockType = 0x00;
```

Comment:
This is a trivial tip, but it is recommended not to use 0 as the ID to be used for RLP.
The zero is converted to `0x80` and it sometimes gives a confusion.